### ISSUE TITLE
Add omnibox item for Brave AI when enabled

### ISSUE THREAD

**Issue by bbondy** [brave/brave-browser#30556]
When the Brave Chat flag is on (At the time of writing it is on in Nightly):
- Have a URL bar suggestion entry 
  - it should not default to submit to it unless explicitly selected (e.g. if the user types something and hits enter only it should never go to this item)
- It should lead to the sidebar opening with no page context (AutocompleteProvider).
- If the user is not opted in, it should show the disclaimer first, but if the user does end up opting in, then it should submit the pending query.
- Add another setting sub-item to "Show autocomplete suggestions in address bar" in brave://settings/appearance for Leo chat

**Comment by aguscruiz** [comment]
The first iteration of the omnibar is here:
https://www.figma.com/file/h7fKSJF2P556bZWfyLTfwD/Brave-AI?type=design&node-id=2839%3A53706&mode=design&t=ssHFW0i6y9aMI2eL-1

Also the Settings screen to enable here:
https://www.figma.com/file/h7fKSJF2P556bZWfyLTfwD/Brave-AI?type=design&node-id=1105%3A28709&mode=design&t=ssHFW0i6y9aMI2eL-1

**Comment by petemill** [comment]
@aguscruiz that Figma link isn't working anymore after re-organization. Would you mind updating it?

**Comment by petemill** [comment]
Perhaps it should only be present when a search suggestion is also presented in the omnibox. In other words, never for an exact url match.

The way this _could_ be implemented, is to:
- Get the active Browser
- Ask the sidebar to open the AI Chat panel for the sidebar (if it's not open already)
- Get the `AIChatTabHelper` class for the active Tab in the Browser and add a function to that class named `SetPendingHumanMessage`. If `HasUserOptedIn()` is `true`, then the message can be sent immediately via `MakeAPIRequestWithConversationHistoryUpdate(...)`. If `HasUserOptedIn()` is `false` then that message can be saved to a new instance variable, `std::string pending_human_input_`, which can be sent later when `OnUserOptedIn()` is called.

**Comment by petemill** [comment]
@aguscruiz @sangwoo108 I think the design (and implementation) misunderstood this task:

> Add another setting sub-item to "Show autocomplete suggestions in address bar" in brave://settings/appearance for Leo chat

Instead of that text showing on the Leo subpage, Brave Leo should show in brave://settings/appearance underneath the _extisting_ item:

<img width="752" alt="image" src="https://github.com/brave/brave-browser/assets/741836/96f9300d-ce3e-4090-b341-d92deca4c24d">

😄

**Comment by bbondy** [comment]
It was mentioned in the issue comment 0 but it's hard to know what the source of truth is. Confirming that the screenshot Pete posted above is the correct location.

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using 

```
Brave | 1.61.5 Chromium: 118.0.5993.54 (Official Build) nightly (x86_64)
-- | --
Revision | 0787df601ff078d4344102d73a7be56d520137b0
OS | macOS Version 11.7.10 (Build 20G1427)
```


## Default (un-opted-in)
<details>

</details>

## Opted-in, Leo open
<details>

</details>


## Opted-in, Leo closed
<details>

</details>

## Brave Search promotion
<details>

</details>

**Comment by GeetaSarvadnya** [comment]
Verification INPROGRESS on

```
Brave | 1.60.88 Chromium: 118.0.5993.70 (Official Build) beta (64-bit)
-- | --
Revision | ff9150ac5dd9934a7f431ddf478ad3e45ae68546
OS | Windows 10 Version 22H2 (Build 19045.3570)
```

**Comment by Uni-verse** [comment]
This feature is still in progress on Android. Is it not available for testing at this time. 

@kjozwiak @hffvld

**Comment by kjozwiak** [comment]
> This feature is still in progress on Android. Is it not available for testing as of 1.60 release.
> 
> @kjozwiak @hffvld

Thanks @Uni-verse, we can leave the `OS/Android` label but can skip the verification/checks. The above will always include in our query re: remaining verifications but we can ignore 👍

**Comment by btlechowski** [comment]
Verified with



## `Show autocomplete suggestions in address bar` is `Enabled` (Default)

## Default (un-opted-in) - `PASSED`
<details>

### Steps:

1. installed `1.60.x`
2. launched Brave
3. opened `brave://flags` and set `AI chat` to `Enabled`
4. clicked `Relaunch`
5. typed a query: `is pluto a planet?`
6. confirmed `Ask Leo` appeared below Omnibox autocomplete line
7. confirmed selection remained on the original search query
8. used the `down` arrow-key to select the `is pluto a planet? - ask Leo` entry
9. pressed `return`
10. confirmed `Brave Leo` opened in the sidebar, with an `Accept and begin` opt-in button
11. pressed the button
12. confirmed the query of `is pluto a planet?` "teletyped" in with an answer

example | example | example | example
---------|---------|----------|---------
![image](https://github.com/brave/brave-browser/assets/34715963/b9991981-1e5f-4605-b721-b205dab3d860)|![image](https://github.com/brave/brave-browser/assets/34715963/7d67bda6-37c2-4ba2-b9a8-f0acfa8f58f8)|![image](https://github.com/brave/brave-browser/assets/34715963/0d3de872-7229-4f1d-934c-1f0966d256d1)|![image](https://github.com/brave/brave-browser/assets/34715963/6b1f222d-80b6-45ef-9d22-f78624c7b3bd)


</details>

## Opted-in, Leo open - `PASSED`
<details>

1. installed `1.60.x`
2. launched Brave
3. opened `brave://flags` and set `brave://flags/#brave-ai-chat` to `Enabled`
4. clicked `Relaunch`
5. clicked on `Brave Leo`
6. clicked on `Accept and begin`
7. entered `why are socks smelly?` in the Omnibox/URL bar
8. confirmed focus on URL bar, with selection on `why are socks smelly?`
9. pressed the `down arrow` key to select `why are socks smelly? - Ask Leo`
10. pressed `return`

### Confirmed my query was sent to the open `Brave Leo` tab, where it was answered

![image](https://github.com/brave/brave-browser/assets/34715963/c706f853-ca1d-4191-a1c9-eabe36049aec)|![image](https://github.com/brave/brave-browser/assets/34715963/00717033-debc-4be3-9157-0f5770b5effc)|![image](https://github.com/brave/brave-browser/assets/34715963/14c30a3d-ee52-4ab0-a93b-fe4ce29a4534)|![image](https://github.com/brave/brave-browser/assets/34715963/f82af962-2411-4e86-a16b-dc33b0cf2c1e)|![image](https://github.com/brave/brave-browser/assets/34715963/dbbe5f84-023e-4137-9fb0-73613a247aa0)
--|--|--|--|--



</details>


## Opted-out, Leo closed, `Show sidebar on mouseover` - `PASSED`
<details>

### Steps:
1. installed `1.60.x`
2. launched Brave
3. set `brave://flags/#brave-ai-chat` to `Enabled`
4. clicked `Relaunch`
5. context-clicked on the sidebar
6. chose `On mouseover` for `Show Sidebar`
7. entered `why are some people allergic to bee stings?` in the Omnibox/URL bar
8. confirmed `why are some people allergic to bee stings? - Ask Leo` menu-item option appeared below the `why are some people allergic to bee stings? - Brave Search` entry
9. confirmed focus was retained on my original input
10. pressed the `down arrow` key to select `why are some people allergic to bee stings? - Ask Leo` option
11. pressed `return` (also used mouse)
12. confirmed `About Brave Leo` panel with `Accept and begin` button opened
13. clicked on `Accept and begin`

### Confirmed the response to my query was "teletyped" in

![image](https://github.com/brave/brave-browser/assets/34715963/dbffea73-d6ba-4131-82f9-1802cb1f723c)|![image](https://github.com/brave/brave-browser/assets/34715963/41119dc3-3b00-4e29-ad95-d21a983e05de)|![image](https://github.com/brave/brave-browser/assets/34715963/e7562a1a-be3f-4d24-885a-570e855a5380)
--|--|--


</details>

## Brave Search promotion - `PASSED`


<details>

NOTE: this was verified using `1.61.x` because I couldn't get the `Brave Search` experiment to show up on `beta`

### Steps:
1. installed `1.61.x`
15. launched Brave using `--variations-server-url=https://variations.bravesoftware.com/seed`
16. opened `brave://flags` and set `AI chat` to `Enabled`
17. clicked `Relaunch`
18. changed from `Brave` to `DuckDuckGo` search engine in `brave://settings/search`
19. typed a search query: `crocodile vs. alligator`
20. confirmed `Ask Leo` appeared below Omnibox autocomplete line, and above `Brave Search` promo
21. confirmed selection remained on the `DuckDuckGo` search query
22. used the `down` arrow-key to select the `crocodile vs. alligator - Ask Leo`
23. pressed `return`
24. confirmed `Brave Leo` opened in the sidebar, with an `Accept and begin` opt-in button
25. pressed the button
26. confirmed the search query of `crocodile vs. alligator` "teletyped" in with an answer








</details>

---

## `Show autocomplete suggestions in address bar` is `Disabled` - `PASSED`
<details>

### Steps:
1. installed `1.60.x`
2. launched Brave
3. opened `brave://flags` and set `AI chat` to `Enabled`
4. opened `brave://settings/appearance`
5. unchecked `Brave Leo AI Assistant` beneath `Show autocomplete suggestions in address bar`
6. opened a new-tab window
7. typed `generic search term` 
8. looked at the Omnibox/URL bar

### Confirmed no `Brave Leo` Omnibox suggestion when entering search items in the URL bar

![image](https://github.com/brave/brave-browser/assets/34715963/106b35ce-ab83-4f83-844d-2e3340bd8d2a)|![image](https://github.com/brave/brave-browser/assets/34715963/385de032-eed2-4178-9a8c-90ef73a346dd)


</details>
